test/model/modelInfo/ModelInfo.test.ts

```diff
@@ -0,0 +1,148 @@
+import xml2js from "xml2js";
+import ModelInfo, {
+  ModelInfoXml,
+} from "../../../src/model/modelInfo/ModelInfo";
+import SimpleElement from "../../../src/model/modelInfo/SimpleElement";
+import TypeInfo from "../../../src/model/modelInfo/TypeInfo";
+import { ElementXml } from "../../../src/model/modelInfo/ElementFactory";
+import ChoiceElement from "../../../src/model/modelInfo/ChoiceElement";
+import ListElement from "../../../src/model/modelInfo/ListElement";
+
+describe("ModelInfo", () => {
+  describe("constructor", () => {
+    it("should initalize variables correctly", () => {
+      const elements = [new SimpleElement("foo", "bar", "baz")];
+      const typeInfo = new TypeInfo(
+        "name",
+        "namespace",
+        "baseTypeName",
+        "baseTypeNamespace",
+        elements
+      );
+
+      const modelInfo = new ModelInfo("name", "version", [typeInfo]);
+      expect(modelInfo.name).toBe("name");
+      expect(modelInfo.version).toBe("version");
+      expect(modelInfo.types).toStrictEqual([typeInfo]);
+    });
+  });
+
+  describe("createTypeInfo()", () => {
+    let input: ModelInfoXml;
+
+    beforeEach(() => {
+      const element1: ElementXml = {
+        $: {
+          name: "memberName1",
+          elementType: "elNamespace1.elTypeName1",
+        },
+      };
+      const typeInfoXml = {
+        $: {
+          name: "typeInfoName",
+          namespace: "typeNamespace",
+          baseType: "baseNs.baseType",
+        },
+        "ns4:element": [element1],
+      };
+
+      input = {
+        "ns4:modelInfo": {
+          $: {
+            name: "modelInfoName",
+            version: "modelInfoVersion",
+          },
+          "ns4:typeInfo": [typeInfoXml],
+        },
+      };
+    });
+
+    test("creates a ModelInfo from XML representation", () => {
+      const modelInfo: ModelInfo = ModelInfo.createModelInfo(input);
+      expect(modelInfo.name).toBe("modelInfoName");
+      expect(modelInfo.version).toBe("modelInfoVersion");
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5966907)